langchain-core/src/language_models/tests/chat_models.test.ts

```diff
@@ -36,3 +37,68 @@ test("Test ChatModel uses callbacks with a cache", async () => {
   expect(response.content).toEqual(response2.content);
   expect(response2.content).toEqual(acc);
 });
+
+test("Test ChatModel uses callbacks with a cache", async () => {
+  const model = new FakeChatModel({
+    cache: true,
+  });
+  let acc = "";
+  const response = await model.invoke("Hello there!");
+  const response2 = await model.invoke("Hello there!", {
+    callbacks: [
+      {
+        handleLLMNewToken: (token: string) => {
+          console.log(token);
+          acc += token;
+        },
+      },
+    ],
+  });
+  expect(response.content).toEqual(response2.content);
+  expect(response2.content).toEqual(acc);
+});
+
+test("Test ChatModel withStructuredOutput", async () => {
+  const model = new FakeListChatModel({
+    responses: [`{ "test": true, "nested": { "somethingelse": "somevalue" } }`],
+  }).withStructuredOutput({
```

Comment:
We should add generics in tests so if they ever break it'll fail build and we'll know